http-verbs-common/src/main/scala/uk/gov/hmrc/http/client/HttpClientV2Impl.scala

```diff
@@ -284,30 +286,32 @@ class ExecutorImpl(
   private def executeHooks(
     isStream        : Boolean,
     request         : WSRequest,
-    hookDataF       : Future[Option[HookData]],
-    auditedResponseF: Future[HttpResponse] // play-auditing expects the body to be a String
+    hookDataF       : Future[Body[Option[HookData]]],
+    auditedResponseF: Future[ResponseData]
   )(implicit
     hc: HeaderCarrier,
     ec: ExecutionContext
   ): Unit = {
     def denormalise(hdrs: Map[String, Seq[String]]): Seq[(String, String)] =
       hdrs.toList.flatMap { case (k, vs) => vs.map(k -> _) }
 
-    def executeHooksWithHookData(hookData: Option[HookData]) =
+    def executeHooksWithHookData(hookData: Body[Option[HookData]]) =
       hooks.foreach(
         _.apply(
           verb      = request.method,
           url       = new URL(request.url),
-          headers   = denormalise(request.headers),
-          body      = hookData,
+          request   = RequestData(
+                        headers = denormalise(request.headers),
+                        body    = hookData
+                      ),
           responseF = auditedResponseF
         )
       )
 
     hookDataF.onComplete {
       case Success(hookData) => executeHooksWithHookData(hookData)
       case Failure(e)        => // this is unlikely, but we want best attempt at auditing
-                                executeHooksWithHookData(None)
+                                executeHooksWithHookData(Body.Omitted)
```

Comment:
This is the only place we use `Omitted` currently. The intention of `Omitted` is if clients decide there's no reason to include the payload in audits. Should we add an `Unavailable` state to distinguish?

Reply:
👍 I'm generally in favour of enumerating as many of these states as we can think of. Consumers are free to map them to a smaller set of states if they're not valuable to them.